### PR TITLE
UCP: Fix RNDV truncated on receiver

### LINKED ISSUE #1538: Assertion in udrcx/test_ucp_tag_xfer.iov_exp_truncated/1
Happens in:
http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/label=hpc-test-node2,worker=2/3655/consoleFull

12:11:22 [     INFO ] 0 1581x10^1 500x10^2 158x10^3 50x10^4 15x10^5 5x10^6 [hpc-test-node2:11583:0]        rndv.c:379  Assertion `rreq->recv.length != 0' failed

The problem is related to handling of truncated case. It is possible that RNDV RTS message arrives while receiver waits for 0-byte message.  It should be handled ok rather than having an assert.
The problem is most probably a rare case in jenkins testing, because the buffer lengths are random (calculated in test_ucp_tag_xfer::test_xfer).

### PR DESCRIPTION
**Pull request by brminich**
Fixes #1538

### CODE REVIEW

**Comment by yosefe** [comment]
not relevant for v1.2

**Comment by yosefe** [comment]
@brminich pls add unit test

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1726/ for details.

**Comment by brminich** [comment]
@yosefe, modified existing test, because it is excess  (the same is tested in test_ucp_tag_xfer)

File: test/gtest/ucp/test_ucp_tag_match.cc [hunk 1]
```diff
@@ -515,8 +514,9 @@ UCS_TEST_P(test_ucp_tag_match, rndv_truncated, "RNDV_THRESH=1048576") {
     /* receiver - get the RTS and put it into unexpected */
     short_progress_loop();
 
-    /* receiver - issue a receive request, match it with the RTS and perform rndv get */
-    status = recv_b(&recvbuf[0], (recvbuf.size())/2, DATATYPE, 0x1337, 0xffff, &info);
+    /* receiver - issue a receive request with zero length,
+     * no assertions should occur */
+    status = recv_b(NULL, 0, DATATYPE, 0x1337, 0xffff, &info);
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_tag_match.cc]: can we keep the old test, so we eventually test both with zero and non-zero receive length?

Comment (brminich) [inline test/gtest/ucp/test_ucp_tag_match.cc]: Non-zero length is already tested in test_ucp_tag_xfer (with half size like it was done here)

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1728/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3666/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3668/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3678/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:bgate:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1742/ for details.

**Comment by yosefe** [comment]
bot:mlx:retest

**Comment by yosefe** [comment]
bot:bgate:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1752/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3690/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@brminich pls fix for v1.2 as well if needed

**Pull request merged**